src/stdlib_quadrature_gauss.f90

```diff
@@ -0,0 +1,115 @@
+submodule (stdlib_quadrature) stdlib_quadrature_gauss
+    use stdlib_kinds, only: sp, dp, qp
+    use stdlib_functions, only: legendre, dlegendre
+    implicit none
+
+    real(dp), parameter :: PI = acos(-1._dp)
+    real(dp), parameter :: tolerance = 4._dp * epsilon(1._dp)
+    integer, parameter :: newton_iters = 100
+
+contains
+
+    pure module subroutine gauss_legendre_fp64 (x, w, interval)
+        real(dp), intent(out) :: x(:), w(:)
+        real(dp), intent(in), optional :: interval(2)
+
+        associate (N => size(x)-1 )
+        select case (N)
+            case (0)
+                x = 0._dp
+                w = 2._dp
+            case (1)
+                x = [-sqrt(1._dp/3._dp), sqrt(1._dp/3._dp)]
+                w = [1._dp, 1._dp]
```

Comment:
```
x(1) = -sqrt(1.0_dp/3.0_dp)
x(2) = -x(1)
w = 1
```
Perhaps this is not worth bothering about. Hopefully, the compiler does not generate temporaries.